startleft/config/default-ctf-mapping.yaml

```diff
@@ -181,14 +189,199 @@ components:
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   elasticache
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::ElastiCache')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   CD-GUARDDUTY
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::GuardDuty')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   CD-INSPECTOR
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::Inspector')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   CD-MACIE
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::Macie')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   CD-AWS-NETWORK-FIREWALL
+    name:   {$path: "_key"}
+    $source: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::NetworkFirewall::Firewall')]"}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   redshift
+    name:   {$path: "_key"}
+    $source: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::Redshift::Cluster')]"}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   CD-SES
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::SES')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   sns
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::SNS')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   step-functions
+    name:   {$path: "_key"}
+    $source: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::StepFunctions::StateMachine')]"}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+
+  - id:     {$format: "{name}"}
+    type:   CD-WAF
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::WAF')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   kinesis-data-analytics
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::KinesisAnalytics')]"}}
+    parent:  public-cloud
+    tags:
+      - { $path: "Type" }
+
+  - id:     {$format: "{name}"}
+    type:   kinesis-data-analytics
+    name:   {$path: "_key"}
+    $source: {$singleton: {$root: "Resources|squash(@)[?starts_with(Type, 'AWS::Kinesis::')]"}}
```

Comment:
I'm not really sure about how this works, but it seems that a AWS::Kinesis::Stream is just a data flow and the real work is done in the AWS::KinesisAnalytics::Application and this is where the stream is used as an input. If this is true maybe we should not map AWS::Kinesis:: because doing this we will see two kinesis-data-analytics in the threatmodel, but just one of them makes sense.

Reply:
My idea was that if any Kinesis component is present it should be mapped into one single kinesis-data-analytics component. The $singleton checks if the component already exists in the OTM, so if you find a Kinesis stream and a Kinesis Application only one kinesis-data-analytics component will be created.